storage/src/robocorp/storage/_storage.py

```diff
@@ -22,8 +33,12 @@
 )
 
 
-class AssetNotFound(KeyError):
-    """Raised when the queried asset couldn't be found."""
+class AssetNotFound(RequestsHTTPError):
+    """No asset with given name/id found."""
+
+
+class AssetUploadFailed(RuntimeError):
+    """There was an unexpected error while uploading an asset."""
```

Comment:
Let me know if punctuation should be missing on docstrings as well, but according to [PEP8](https://peps.python.org/pep-0008/) these look aligned now.